src/m365/tenant/commands/auditlog/auditlog-report.ts

```diff
@@ -0,0 +1,272 @@
+          };
+
+          return request.post(requestOptions);
+        }
+        else {
+          return Promise.resolve();
+        }
+      });
+  }
+
+  private getAuditContentList(args: CommandArgs, logger: Logger): Promise<AuditContentList[]> {
+    if (this.verbose) {
+      logger.logToStderr(`Start listing Audit Content URL`);
+    }
+    let SubscriptionListEndpoint: string = `activity/feed/subscriptions/content?contentType=${(<any>AuditContentTypes)[args.options.contentType]}&PublisherIdentifier=${this.tenantId}`;
+
+    if (typeof args.options.startTime !== 'undefined') {
+      const parsedEndDate: any = new Date(args.options.startTime);
+      parsedEndDate.setDate(parsedEndDate.getDate() + 1);
+      SubscriptionListEndpoint += `&starttime=${escape(args.options.startTime)}&endTime=${escape(parsedEndDate.toISOString())}`;
+    }
+    const requestOptions: any = {
+      url: `${this.serviceUrl}/${this.tenantId}/${SubscriptionListEndpoint}`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    return request.get<AuditContentList[]>(requestOptions)
+  }
+
+  private getBatchedPromises(AuditContentLists: AuditContentList[], batchSize: number): void {
+    for (let i: number = 0; i < AuditContentLists.length; i += batchSize) {
+      const PromiseRequestBatch: Promise<AuditlogReport[]>[] = AuditContentLists.slice(i, i + batchSize < AuditContentLists.length ? i + batchSize : AuditContentLists.length).map((AuditContentList: AuditContentList) => {
+        return this.getAuditLogReportforSingleContentURL(AuditContentList.contentUri);
+      })
+      this.batchedPromises.push(PromiseRequestBatch);
+    }
+  }
+
+  private getBatchedAuditlogData(logger: Logger, batchNumber: number, resolve: () => void, reject: (err: any) => void): void {
+
+    if (this.verbose) {
+      logger.logToStderr(`Starting Batch : ${batchNumber}`);
+    }
+    Promise
+      .all(this.batchedPromises[batchNumber])
+      .then((data: any) => {
+        this.completeAuditReports.push(data);
+        if (batchNumber < this.batchedPromises.length - 1) {
+          this.getBatchedAuditlogData(logger, ++batchNumber, resolve, reject)
+        }
+        else {
+          resolve();
+        }
+      }, err => { return Promise.reject(err) });
+  }
+
+  private getAuditLogReportforSingleContentURL(auditURL: string): Promise<AuditlogReport[]> {
+    const requestOptions: any = {
+      url: auditURL,
+      headers: {
+        accept: 'application/json;'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get<AuditlogReport[]>(requestOptions);
+  }
+
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-c, --contentType <contentType>',
+        description: 'Audit content type of logs to be retrieved, should be one of the following: AzureActiveDirectory, Exchange, SharePoint, General, DLP'
```

Comment:
We should add these values to `autocomplete`. See https://github.com/pnp/cli-microsoft365/blob/fb6b9f31ac97ce9dc4a00dd1e6377e82f7a61fdc/src/Command.ts#L151 for an example